scipy/stats/tests/test_contingency.py

```diff
@@ -196,3 +196,62 @@ def test_chi2_contingency_bad_args():
     obs = np.empty((0, 8))
     assert_raises(ValueError, chi2_contingency, obs)
 
+
+def test_bad_association_args():
+    # Invalid Chi Squared Stat Value Type
+    assert_raises(ValueError, association, [[1, 2], [3, 4]], "x", "tschuprow")
+    # Invalid Test Statistic
+    assert_raises(ValueError, association, [[1, 2], [3, 4]], 2.55, "X")
+    # Invalid array shape
+    assert_raises(ValueError, association, [[[1, 2]], [[3, 4]]], 2.55, "cramer")
+
+
+def test_cramersv():
+    # 2d Array
+    obs = np.array([[12, 13, 14, 15, 16],
+                    [17, 16, 18, 19, 11],
+                    [9, 15, 14, 12, 11]])
+    chi2 = chi2_contingency(obs)[0]
+    a = association(obs, chi2_stat=chi2,
+                    stat='v')
+
+    correct = 0.09222412010290792
+    assert_approx_equal(a, correct)
```

Comment:
These days, the preferred function is `assert_allclose` (see the note in the [`assert_approx_equal`](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_approx_equal.html) docstring):
```suggestion
    assert_allclose(a, correct)
```